### ISSUE TITLE
empty src: local('')

### ISSUE THREAD

**Issue by jonas-herkelmedia** [majodev/google-webfonts-helper#124]
The `@font-face` has always an empty string for `src: local('')`. This was not the case a few months ago.

**Comment by paulca55** [comment]
I came here to report the same issue. It looks like Google have removed the local() function due to lot of users having problems with local fonts not correctly representing what the user should see.

https://twitter.com/googlefonts/status/1323363090969333762

**Comment by jonas-herkelmedia** [comment]
Ok now it makes sense. The local fonts are often outdated, but i heard there is a problem with an empty local() function on Android devices. Isn't it best to remove it completely?

https://github.com/majodev/google-webfonts-helper/issues/84

**Comment by witcher83** [comment]
The `src: local('')` could be filled with the font-family value. In this way, the browser will check if the font is already present in the local machine and avoid downloading it.
For example, in the case of the "Raleway" font, it should be: `src: local('Raleway')`

**Comment by gsauthof** [comment]
> The local fonts are often outdated, but i heard there is a problem with an empty local() function on Android devices. Isn't it best to remove it completely?

FWIW, for my use cases local outdated fonts aren't a big concern.
You could make it optional.
AFAICS, simply removing it definitely is an improvement over having an empty `local('')` declaration.

**Comment by mangelozzi** [comment]
For example `Roboto`, this is what local should be set to:
(If set to `''`, then it will use network fonts always. If set to `Roboto`, will always use the regular font-weight version.

```
/* ----------------------------------------------------------------------------
   ROBOTO
*/
/* roboto-300 - latin */
@font-face {
  font-family: 'Roboto';
  font-style: normal;
  font-weight: 300;
  src: local('Roboto-Light'),
       url('/static/fonts/roboto-v29-latin-300.woff2') format('woff2'), /* Chrome 26+, Opera 23+, Firefox 39+ */
       url('/static/fonts/roboto-v29-latin-300.woff') format('woff'); /* Chrome 6+, Firefox 3.6+, IE 9+, Safari 5.1+ */
  font-display: swap;
}
/* roboto-regular - latin */
@font-face {
  font-family: 'Roboto';
  font-style: normal;
  font-weight: 400;
  src: local('Roboto-Regular'),
       url('/static/fonts/roboto-v29-latin-regular.woff2') format('woff2'), /* Chrome 26+, Opera 23+, Firefox 39+ */
       url('/static/fonts/roboto-v29-latin-regular.woff') format('woff'); /* Chrome 6+, Firefox 3.6+, IE 9+, Safari 5.1+ */
  font-display: swap;
}
/* roboto-500 - latin */
@font-face {
  font-family: 'Roboto';
  font-style: normal;
  font-weight: 500;
  src: local('Roboto-Medium'),
       url('/static/fonts/roboto-v29-latin-500.woff2') format('woff2'), /* Chrome 26+, Opera 23+, Firefox 39+ */
       url('/static/fonts/roboto-v29-latin-500.woff') format('woff'); /* Chrome 6+, Firefox 3.6+, IE 9+, Safari 5.1+ */
  font-display: swap;
}
/* roboto-700 - latin */
@font-face {
  font-family: 'Roboto';
  font-style: normal;
  font-weight: 700;
  src: local('Roboto-Bold'),
       url('/static/fonts/roboto-v29-latin-700.woff2') format('woff2'), /* Chrome 26+, Opera 23+, Firefox 39+ */
       url('/static/fonts/roboto-v29-latin-700.woff') format('woff'); /* Chrome 6+, Firefox 3.6+, IE 9+, Safari 5.1+ */
  font-display: swap;
}
```

**Comment by osworx** [comment]
Maybe read this why filling the local variable with a name is not the best idea:
https://www.paulirish.com/2009/bulletproof-font-face-implementation-syntax/

What I've done is described below with the code I'm using - works perfectly.
Ordering of the fonts is from "Super Modern" > "Forgetable" browser from the bottom to top (the way the fonts are called) - I think how this project orders the fonts should be also changed ..

Assuming your fonts are 1 folder higher than the css itself (**fonts/**).
Comments can be deleted (just here for demonstration).
And - finally - the css (especially when many fonts are used) should be minimized (save approx- 30%) :)

```css /* open-sans-regular - vietnamese_latin_hebrew_greek_cyrillic */
@font-face {
  font-family: 'Open Sans';
  font-style: normal;
  font-weight: 400;
  font-display: swap;
  src: url('fonts/open-sans-v34-vietnamese_latin_hebrew_greek_cyrillic-regular.eot'); /* IE9 Compat Modes */
  src: local('😡');
  src: url('fonts/open-sans-v34-vietnamese_latin_hebrew_greek_cyrillic-regular.eot?#iefix') format('embedded-opentype'), /* IE6-IE8 */
       url('fonts/open-sans-v34-vietnamese_latin_hebrew_greek_cyrillic-regular.svg#OpenSans') format('svg'), /* Legacy iOS */
       url('fonts/open-sans-v34-vietnamese_latin_hebrew_greek_cyrillic-regular.ttf') format('truetype'), /* Safari, Android, iOS */
       url('fonts/open-sans-v34-vietnamese_latin_hebrew_greek_cyrillic-regular.woff') format('woff'), /* Modern Browsers */
       url('fonts/open-sans-v34-vietnamese_latin_hebrew_greek_cyrillic-regular.woff2') format('woff2'); /* Super Modern Browsers */
}
```

**Comment by DanielRuf** [comment]
@majodev please remove the `local('')`part.

Users of Adobe Photoshop told me, that they see only broken / weird characters due to a locally installed font with no name.

Also when I copy the code from your Google Webfonts Helper, I always manually remove this part and recommend the same to others.

Others still blindly copy this code and don't understand why they get some foreign-looking glyphs instead of the expected font.

**Comment by DanielRuf** [comment]
> The `src: local('')` could be filled with the font-family value. In this way, the browser will check if the font is already present in the local machine and avoid downloading it.
> For example, in the case of the "Raleway" font, it should be: `src: local('Raleway')`

How do you know if they have the same font version that you use so that all glyphs can be displayed as expected? You can't. Often these local versions are totally outdated, are missing features like ligatures and contain bugs. No user will update the locally installed version when your website looks broken. They often will say, that your website / fonts on your server are broken.

**Comment by majodev** [comment]
Will be removed with the next release.

**Comment by DanielRuf** [comment]
> Will be removed with the next release.

Perfect. If needed, here is a small PR: https://github.com/majodev/google-webfonts-helper/pull/166

I guess these server parts are also relevant: https://github.com/search?q=repo%3Amajodev%2Fgoogle-webfonts-helper+localName&type=code

**Comment by majodev** [comment]
Fixed in [v1.1.0](https://github.com/majodev/google-webfonts-helper/releases/tag/v1.1.0).
